core/src/test/scala/integration/kafka/api/TransactionsExpirationTest.scala

```diff
@@ -96,14 +101,28 @@ class TransactionsExpirationTest extends KafkaServerTestHarness {
     waitUntilTransactionalStateExists()
     waitUntilTransactionalStateExpires()
 
-    // Start a new transaction and attempt to send, which will trigger an AddPartitionsToTxnRequest, which will fail due to the expired transactional ID.
-    producer.beginTransaction()
-    val failedFuture = producer.send(TestUtils.producerRecordWithExpectedTransactionStatus(topic1, 3, "1", "1", willBeCommitted = false))
-    TestUtils.waitUntilTrue(() => failedFuture.isDone, "Producer future never completed.")
-
-    org.apache.kafka.test.TestUtils.assertFutureThrows(failedFuture, classOf[InvalidPidMappingException])
-    producer.abortTransaction()
+    try {
+      // Start a new transaction and attempt to send, which will trigger an AddPartitionsToTxnRequest,
+      // which will fail due to the expired transactional ID, causing a fatal error.
+      producer.beginTransaction()
+      val failedFuture = producer.send(TestUtils.producerRecordWithExpectedTransactionStatus(topic1, 3, "1", "1", willBeCommitted = false))
+      TestUtils.waitUntilTrue(() => failedFuture.isDone, "Producer future never completed.")
+
+      org.apache.kafka.test.TestUtils.assertFutureThrows(failedFuture, classOf[InvalidPidMappingException])
+
+      // Since this is a fatal error, attempting to abort should fail.
+      producer.abortTransaction()
+    } catch {
+      case _: KafkaException =>
```

Comment:
Should we assert we get an error on abort and not have the reinitialization in a catch block? We want to ensure it happens.